src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -257,3 +260,31 @@ export const extractRegisteredAuthTypeCredentials = (
 
   return registeredCredentials;
 };
+
+interface DataSourceOptionGroupLabel {
+  [key: string]: DataSourceGroupLabelOption;
+}
+
+export const dataSourceOptionGroupLabel = deepFreeze<Readonly<DataSourceOptionGroupLabel>>({
+  opensearchCluster: {
+    id: 'opensearchClusterGroupLabel',
+    label: 'OpenSearch cluster',
+    isGroupLabel: true,
+  },
+  // TODO: add other group labels if needed
+});
+
+const createApplicationService = () => {
```

Comment:
can we use the existing https://github.com/opensearch-project/OpenSearch-Dashboards/blob/36c25aed92dbc583e02e0569f42fb8476de7ad67/src/plugins/data/public/services.ts#L37

Reply:
sure, updated to use  existing setter  and getter